File: RoboLabWP/RoboLabWPComp/Model/Basic Types/MZMotion.cpp
#include "pch.h"
#include "MZMotion.h"


MZMotion::MZMotion(double speed, MZDirection direction)
{
	_speed = speed;
    _direction = direction;
    _timesUsed = 0;
}


MZMotion::~MZMotion(void)
{
}

void MZMotion::use(void)
{
	_timesUsed++;
}

int MZMotion::timesUsed(void)
{
	return _timesUsed;
}

File: RoboLabWP/RoboLabWP/Model/Basic Types/MZMotion.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RoboLabWP.Model.Basic_Types
{
    public class MZMotion
    {
        public double _speed;
        public MZDirection _direction;
        int _timesUsed;

        public MZMotion(double speed, MZDirection direction)
        {
            _speed = speed;
            _direction = direction;
            _timesUsed = 0;
        }

        void use()
        {
            _timesUsed++;
        }

        int timesUsed()
        {
            return _timesUsed;
        }
    }
}
